lib/sycamore/sycamore/data/table.py

```diff
@@ -0,0 +1,164 @@
+from collections import OrderedDict
+from dataclasses import dataclass, field
+from typing import Any, Optional
+import xml.etree.ElementTree as ET
+
+from sycamore.data import BoundingBox
+from PIL import Image, ImageDraw
+import numpy as np
+from pandas import DataFrame
+
+
+# This data model is similar to that used by Textract and TableTransformers.
+# It is intended to be a common intermediate representation for a variety of
+# table structure recognition models.
+@dataclass(frozen=True)
+class TableCell:
+    content: str
+    rows: list[int]
+    cols: list[int]
+    is_header: bool = False
+    bbox: Optional[BoundingBox] = None
+    # Model/format specific properties
+    properties: dict[str, Any] = field(default_factory=dict)
+
+
+class Table:
+    def __init__(self, cells: list[TableCell], caption: Optional[str] = None):
+        self.cells: list[TableCell] = cells
+        self.caption = caption
+        self.num_rows = max(max(c.rows) for c in self.cells) + 1
+        self.num_cols = max(max(c.cols) for c in self.cells) + 1
+
+    def __eq__(self, other):
+        if type(other) is not type(self):
+            return False
+
+        if self.num_rows != other.num_rows or self.num_cols != other.num_cols:
+            return False
+
+        # TODO: Yikes! this is expensive. Do the sorting in init or something.
+        if self.sorted_cells() != other.sorted_cells():
+            return False
+
+        return True
+
+    def __hash__(self):
+        return hash((self.num_rows, self.num_cols, self.cells))
```

Comment:
Hash(cells) should be sufficient, no?  And only if the order is canonical, right?  Otherwise, there's a divergence between eq(x,y) and eq(hash(x),hash(y)).

Reply:
Yeah, you are right. I will update.